### ISSUE TITLE
Creative Mode middle click on large wooden gear crashes game and breaks world

### ISSUE THREAD

**Issue by SpectrumCode** [anegostudios/VintageStory-Issues#4314]
### Game Version

1.20.0-pre.6

### Platform

Windows

### Modded

Vanilla

### SP/MP

Singleplayer

### Description

Middle clicking a large wooden gear in creative mode crashes the game. Game crashes attempting to reload world after first error.

### How to reproduce

1. Place large wooden gear.
2. Middle click wooden gear.
3. Crash 1.
4. Restart game
5. Load World
6. Crash 2. 

### Logs
```
Crash 1 "
Running on 64 bit Windows 10.0.22631.0 with 81830 MB RAM
Game Version: v1.20.0-pre.6 (Unstable)
16/10/2024 3:52:13 PM: Critical error occurred
Loaded Mods: game@1.20.0-pre.6, creative@1.20.0-pre.6, survival@1.20.0-pre.6
System.NullReferenceException: Object reference not set to an instance of an object.
   at Vintagestory.GameContent.EntityShapeRenderer.RenderItem(Single dt, Boolean isShadowPass, ItemStack stack, AttachmentPointAndPose apap, ItemRenderInfo renderInfo) in VSEssentials\EntityRenderer\EntityShapeRenderer.cs:line 477
   at Vintagestory.GameContent.EntityShapeRenderer.RenderHeldItem(Single dt, Boolean isShadowPass, Boolean right) in VSEssentials\EntityRenderer\EntityShapeRenderer.cs:line 473
   at Vintagestory.GameContent.EntityPlayerShapeRenderer.DoRender3DOpaque(Single dt, Boolean isShadowPass) in VSEssentials\EntityRenderer\EntityPlayerShapeRenderer.cs:line 230
   at Vintagestory.Client.NoObf.SystemRenderEntities.OnRenderOpaque3D(Single deltaTime) in VintagestoryLib\Client\Systems\Render\RenderEntities.cs:line 79
   at Vintagestory.Client.NoObf.ClientEventManager.TriggerRenderStage(EnumRenderStage stage, Single dt) in VintagestoryLib\Client\Util\ClientEventManager.cs:line 186
   at Vintagestory.Client.NoObf.ClientMain.TriggerRenderStage(EnumRenderStage stage, Single dt) in VintagestoryLib\Client\ClientMain.cs:line 819
   at Vintagestory.Client.NoObf.ClientMain.MainRenderLoop(Single dt) in VintagestoryLib\Client\ClientMain.cs:line 825
   at Vintagestory.Client.NoObf.ClientMain.MainGameLoop(Single deltaTime) in VintagestoryLib\Client\ClientMain.cs:line 741
   at Vintagestory.Client.GuiScreenRunningGame.RenderToPrimary(Single dt) in VintagestoryLib\Client\MainMenu\Screens\GuiScreenRunningGame.cs:line 207
   at Vintagestory.Client.ScreenManager.Render(Single dt) in VintagestoryLib\Client\ScreenManager.cs:line 695
   at Vintagestory.Client.ScreenManager.OnNewFrame(Single dt) in VintagestoryLib\Client\ScreenManager.cs:line 670
   at Vintagestory.Client.NoObf.ClientPlatformWindows.window_RenderFrame(FrameEventArgs e) in VintagestoryLib\Client\ClientPlatform\GameWindow.cs:line 88
   at OpenTK.Windowing.Desktop.GameWindow.Run()
   at Vintagestory.Client.ClientProgram.Start(ClientProgramArgs args, String[] rawArgs) in VintagestoryLib\Client\ClientProgram.cs:line 328
   at Vintagestory.Client.ClientProgram.<>c__DisplayClass10_0.<.ctor>b__1() in VintagestoryLib\Client\ClientProgram.cs:line 130
   at Vintagestory.ClientNative.CrashReporter.Start(ThreadStart start) in VintagestoryLib\Client\ClientPlatform\ClientNative\CrashReporter.cs:line 93
"

Crash 2 "
Running on 64 bit Windows 10.0.22631.0 with 81830 MB RAM
Game Version: v1.20.0-pre.6 (Unstable)
16/10/2024 3:58:47 PM: Critical error occurred
Loaded Mods: game@1.20.0-pre.6, creative@1.20.0-pre.6, survival@1.20.0-pre.6
System.NullReferenceException: Object reference not set to an instance of an object.
   at Vintagestory.GameContent.EntityShapeRenderer.RenderItem(Single dt, Boolean isShadowPass, ItemStack stack, AttachmentPointAndPose apap, ItemRenderInfo renderInfo) in VSEssentials\EntityRenderer\EntityShapeRenderer.cs:line 533
   at Vintagestory.GameContent.EntityShapeRenderer.RenderHeldItem(Single dt, Boolean isShadowPass, Boolean right) in VSEssentials\EntityRenderer\EntityShapeRenderer.cs:line 472
   at Vintagestory.GameContent.EntityPlayerShapeRenderer.RenderHeldItem(Single dt, Boolean isShadowPass, Boolean right) in VSEssentials\EntityRenderer\EntityPlayerShapeRenderer.cs:line 276
   at Vintagestory.GameContent.EntityPlayerShapeRenderer.DoRender3DOpaque(Single dt, Boolean isShadowPass) in VSEssentials\EntityRenderer\EntityPlayerShapeRenderer.cs:line 197
   at Vintagestory.Client.NoObf.SystemRenderEntities.OnRenderOpaque3D(Single deltaTime) in VintagestoryLib\Client\Systems\Render\RenderEntities.cs:line 92
   at Vintagestory.API.Client.DummyRenderer.OnRenderFrame(Single deltaTime, EnumRenderStage stage) in VintagestoryApi\Client\API\IClientEventAPI.cs:line 83
   at Vintagestory.Client.NoObf.ClientEventManager.TriggerRenderStage(EnumRenderStage stage, Single dt) in VintagestoryLib\Client\Util\ClientEventManager.cs:line 200
   at Vintagestory.Client.NoObf.ClientMain.TriggerRenderStage(EnumRenderStage stage, Single dt) in VintagestoryLib\Client\ClientMain.cs:line 816
   at Vintagestory.Client.NoObf.ClientMain.MainRenderLoop(Single dt) in VintagestoryLib\Client\ClientMain.cs:line 911
   at Vintagestory.Client.NoObf.ClientMain.MainGameLoop(Single deltaTime) in VintagestoryLib\Client\ClientMain.cs:line 740
   at Vintagestory.Client.GuiScreenRunningGame.RenderToPrimary(Single dt) in VintagestoryLib\Client\MainMenu\Screens\GuiScreenRunningGame.cs:line 170
   at Vintagestory.Client.ScreenManager.Render(Single dt) in VintagestoryLib\Client\ScreenManager.cs:line 695
   at Vintagestory.Client.ScreenManager.OnNewFrame(Single dt) in VintagestoryLib\Client\ScreenManager.cs:line 670
   at Vintagestory.Client.NoObf.ClientPlatformWindows.window_RenderFrame(FrameEventArgs e) in VintagestoryLib\Client\ClientPlatform\GameWindow.cs:line 88
   at OpenTK.Windowing.Desktop.GameWindow.Run()
   at Vintagestory.Client.ClientProgram.Start(ClientProgramArgs args, String[] rawArgs) in VintagestoryLib\Client\ClientProgram.cs:line 328
   at Vintagestory.Client.ClientProgram.<>c__DisplayClass10_0.<.ctor>b__1() in VintagestoryLib\Client\ClientProgram.cs:line 130
   at Vintagestory.ClientNative.CrashReporter.Start(ThreadStart start) in VintagestoryLib\Client\ClientPlatform\ClientNative\CrashReporter.cs:line 93
"
```

**Comment by c0riVR** [comment]
Is there anyway to fix the savegame? This exact same Error happened to me

**Comment by radfast** [comment]
The underlying issue was fixed in game version v1.20.0-pre.7.

I don't think this issue with the large wooden gear can have occurred since then, so this can be closed.  (Just tested myself and no crash on picking block, though I notice I do get an invalid invisible item in my hotbar...)

@c0riVR if you have a broken savegame with this issue, and you have updated your game installation to the latest (currently 1.20.0-rc.5) and it is still borken, please mention it here and we must re-open this, or else feel free to create a new issue
